### ISSUE TITLE
Extensible definition dictionary

### ISSUE THREAD

**Issue by ForNeVeR** [ForNeVeR/wpf-math#47]
To avoid polluting our main formula dictionary (i.e. `PredefinedTexFormulas.xml`) with various localized formula names (like [this](https://github.com/ForNeVeR/wpf-math/pull/44#issuecomment-282280393)), I propose to add a dictionary extension mechanism.

Users should be able to provide theirs own dictionary files that will be merged with the existing definitions.

Also, these Russian localized definitions should be extracted from the default dictionary and should became opt-in functionality _(I still want to redistribute them alongside WPF-Math because personally I need those, and I think they're generally useful for people around the world)_.

**Comment by alexreg** [comment]
This might be a reason to get rid of the DSL and define the operators using a builder-syntax in-code... haven't thought it through fully though.

**Comment by ForNeVeR** [comment]
I think that's a separate issue. When planning this, I thought that we could simply split our XML definitions (e.g. `PredefinedTexFormulas.xml`) to multiple files and load them on demand (for example, pass resource names to `TexFormulaParser` constructor or something like this).

**Comment by alexreg** [comment]
I don't think that's nearly as elegant, if I'm honest... keeping everything in code makes the extensibility aspect really easy.

**Comment by ForNeVeR** [comment]
Alright, let's think about that a little more.

I'd like to settle things like this before releasing the "stable" version 1.0, that should have a stable and extensible API we won't accidentally break.

**Comment by alexreg** [comment]
Agreed. I know abandoning the XML definition format may seem like a bit of a waste now, but I think it could be the optimal way forward, especially as far as extensibility is concerned. The DSL could grow to unwieldy proportions otherwise...
